src/lib/ngsi/MetadataVector.cpp

```diff
@@ -87,6 +87,25 @@ std::string MetadataVector::render(const std::string& indent, bool comma)
 
 /* ****************************************************************************
 *
+* MetadataVector::matchFilter -
+*
+*/
+bool MetadataVector::matchFilter(const std::string& mdName, const std::vector<std::string>& metadataFilter)
+{
+
+  /* Metadata filtering only in the case of actual metadata vector not conatining "*" */
+  if ((metadataFilter.size() == 0) || (std::find(metadataFilter.begin(), metadataFilter.end(), NGSI_MD_ALL) != metadataFilter.end()))
+  {
+    return true;
+  }
```

Comment:
The previousValue and actionType metadata has been previously added along with all the other "regular" metadata at processSubscription() function, in particular at this point: https://github.com/telefonicaid/fiware-orion/issues/1298